### PR TITLE
feat(tem) : add ressource for validate tem domain

### LINKED ISSUE #1733: Add a TEM domain check resource
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### New or Affected Resource(s)

* scaleway_XXXXX

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* #0000

### PR DESCRIPTION
**Pull request by jremy42**
close #1733

### CODE REVIEW

File: scaleway/provider.go [hunk 1]
```diff
@@ -149,6 +149,7 @@ func Provider(config *ProviderConfig) plugin.ProviderFunc {
 				"scaleway_lb_route":                            resourceScalewayLbRoute(),
 				"scaleway_registry_namespace":                  resourceScalewayRegistryNamespace(),
 				"scaleway_tem_domain":                          resourceScalewayTemDomain(),
```
Comment (remyleone) [inline scaleway/provider.go:151]: Could this logic be implemented directly inside the `scaleway_tem_domain` ressource?

File: scaleway/testdata/tem-domain-validation-basic.cassette.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    headers:
+      Content-Type:
+      - application/json
+      User-Agent:
+      - scaleway-sdk-go/v1.0.0-beta.7+dev (go1.21.5; darwin; amd64) terraform-provider/develop
+        terraform/terraform-tests
+    url: https://api.scaleway.com/transactional-email/v1alpha1/regions/fr-par/domains
+    method: POST
+  response:
+    body: '{"details":[{"current":3,"quota":3,"resource":"TemDomains"}],"message":"quota(s)
```
Comment (remyleone) [inline scaleway/testdata/tem-domain-validation-basic.cassette.yaml:16]: Not sure this is the message we are looking for

File: scaleway/provider.go (hunk 1 above)
Comment (jremy42) [reply scaleway/provider.go:151]: No, because to validate the domain, we have to make records that depend on the scaleway_tem_domain resource. See => https://registry.terraform.io/providers/scaleway/scaleway/latest/docs/resources/tem_domain#add-the-required-records-to-your-dns-zone. We must wait for the records on the domain before the domain_tem verification.

File: scaleway/testdata/tem-domain-validation-basic.cassette.yaml (hunk 2 above)
Comment (jremy42) [reply scaleway/testdata/tem-domain-validation-basic.cassette.yaml]: This file shouldn't have been pushed; it's an old test.

File: scaleway/sweeper_test.go [hunk 3]
```diff
@@ -93,3 +95,33 @@ func sharedS3ClientForRegion(region scw.Region) (*s3.S3, error) {
 func TestIsTestResource(t *testing.T) {
 	assert.True(t, isTestResource("tf_tests_mnq_sqs_queue_default_project"))
 }
+
+func testSweepTemDomain(_ string) error {
```
Comment (remyleone) [inline scaleway/sweeper_test.go:99]: Why is this function in this file instead of in the tem files ?

File: internal/provider/provider.go [hunk 4]
```diff
@@ -209,6 +209,7 @@ func Provider(config *Config) plugin.ProviderFunc {
 				"scaleway_secret":                              secret.ResourceSecret(),
 				"scaleway_secret_version":                      secret.ResourceVersion(),
 				"scaleway_tem_domain":                          tem.ResourceDomain(),
```
[the remark this answers is not in the document]
Comment (jremy42) [reply internal/provider/provider.go:211]: No, because to validate the domain, we have to make records that depend on the scaleway_tem_domain resource. See => https://registry.terraform.io/providers/scaleway/scaleway/latest/docs/resources/tem_domain#add-the-required-records-to-your-dns-zone. We must wait for the records on the domain before the domain_tem verification.

File: internal/services/tem/domain_validation.go [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+func extractAfterSlash(s string) string {
+	lastIndex := strings.LastIndex(s, "/")
+	if lastIndex == -1 {
+		return s
+	}
+	return s[lastIndex+1:]
+}
+
+func validateDomain(ctx context.Context, d *schema.ResourceData, err error, api *tem.API, region scw.Region) diag.Diagnostics {
```
[the remark this answers is not in the document]
Comment (jremy42) [reply internal/services/tem/domain_validation.go:109]: I find it clearer to have two functions rather than one very heavy function. Easier to read, but I can modify it ?

File: internal/services/tem/domain_validation.go [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+		return diag.FromErr(err)
+	}
+
+	_ = d.Set("validated", domain.Status == "checked")
+
+	return nil
+}
+
+func ResourceDomainValidationDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	_ = ctx
```
Comment (remyleone) [inline internal/services/tem/domain_validation.go:115]: Put it in the signature if you are not using them
```
func ResourceDomainValidationDelete(_ context.Context, d *schema.ResourceData, _ interface{}) diag.Diagnostics {
```

File: scaleway/testdata/data-source-webhosting-basic.cassette.yaml
Comment (remyleone) [inline scaleway/testdata/data-source-webhosting-basic.cassette.yaml:1]: seems that this change does not belong to this PR

**Review by remyleone — APPROVED** [review]
